modules/cli/src/main/scala/scala/cli/exportCmd/MavenProjectDescription.scala

```diff
@@ -0,0 +1,236 @@
+package scala.cli.exportCmd
+
+import coursier.ivy.IvyRepository
+import coursier.maven.MavenRepository
+import coursier.parse.RepositoryParser
+import dependency.{AnyDependency, NoAttributes, ScalaNameAttributes}
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+
+import scala.build.errors.BuildException
+import scala.build.internal.Constants
+import scala.build.internal.Runner.frameworkName
+import scala.build.options.{BuildOptions, Platform, Scope, ShadowingSeq}
+import scala.build.testrunner.AsmTestRunner
+import scala.build.{Logger, Positioned, Sources}
+import scala.cli.ScalaCli
+import scala.cli.exportCmd.POMBuilderHelper._
+import scala.xml.Elem
+
+object POMBuilderHelper {
+  def buildNode(name: String, value: String): Elem =
+    new Elem(
+      null,
+      name,
+      scala.xml.Null,
+      scala.xml.TopScope,
+      minimizeEmpty = false,
+      scala.xml.Text(value)
+    )
+}
+
+final case class MavenProjectDescription(extraSettings: Seq[String], logger: Logger)
+    extends ProjectDescriptor {
+  private val q  = "\""
+  private val nl = System.lineSeparator()
+
+  private def sources(sourcesMain: Sources, sourcesTest: Sources): MavenProject = {
+    val mainSources = ProjectDescriptor.sources(sourcesMain)
+    val testSources = ProjectDescriptor.sources(sourcesTest)
+    MavenProject(
+      mainSources = mainSources,
+      testSources = testSources
+    )
+  }
+
+  private def javaOptionsSettings(options: BuildOptions): MavenProject =
+    MavenProject(
+      settings = Nil
+    )
+
+  private def javacOptionsSettings(options: BuildOptions): List[String] = {
+
+    val javacOptionsSettings =
+      if (options.javaOptions.javacOptions.toSeq.isEmpty) Nil
+      else {
+        val options0 = options
+          .javaOptions
+          .javacOptions
+          .toSeq
+          .map(_.value)
+          .map(o => "\"" + o.replace("\"", "\\\"") + "\"")
+        options0
+      }
+
+    // MavenProject(
+    //   settings = Seq(javacOptionsSettings)
+    // )
+    javacOptionsSettings.toList
+  }
+
+  private def projectArtifactSettings(): MavenProject =
+    // todo: Is it really needed to configure the values, or always use the default?
+    MavenProject(
+      groupId = None,
+      artifactId = None,
+      version = None
+    )
```

Comment:
we only require these when publishing an artifact in Scala CLI.
a Scala CLI project may have them defined if it's configured for publishing.
We should be able to read that.
https://scala-cli.virtuslab.org/docs/reference/directives#publish

As for defaults, I'm not sure if there's any sane ones.